functions/HttpTriggerAPIUsersIdTaskId/__init__.py

```diff
@@ -6,71 +10,124 @@
+                    tasks.completed, tasks.completedDate, CONCAT (users.firstName, ' ', users.lastName) AS "user" 
+                    FROM [dbo].[tasks] JOIN [dbo].[users] 
+                    on [dbo].[tasks].userId = [dbo].[users].userId
+                    WHERE [dbo].[users].userId = ? AND [dbo].[tasks].taskId = ?""")
         cursor.execute(sql_query, param1, param2)
-        logging.info('Executed the query')         
-        data = cursor.fetchall()
-        logging.info(f"Got result: {data}")
-        return data
-#irrespective of results, closes connection
+        logging.debug('Executed the query')          
+        row = cursor.fetchone()
+        logging.info(f"Got result: {row}")
+        if not row:
+            logging.error('No record with the requested parameters')
+            return func.HttpResponse('Task not found', status_code=500)
+        else:
+            columns = [column[0] for column in cursor.description]
+            data = dict(zip(columns, row))
+        return func.HttpResponse(json.dumps(data, default=str), status_code=200, mimetype="application/json")
     finally:
         cursor.close()
         cnxn.close()
-        logging.info('Closed the connection')
+        logging.info('Closed the db connection')   
 
-#POST API method function       
-def post(param):
-    return ('You selected POST method with {param} values. Functionality under construction')
-            # STARTER CODE FOR NEXT SPRINT, SQL QUERY TESTED
-            #     #insert row into tasks table / create a new task
-            #     #createdDate #'20120618 10:34:09 AM' and title/description are hardcoded for sprint1, update with automated date stamp and url params later
-            #     sql_query = ("""INSERT INTO dbo.tasks (userId, title, description, createdDate)
-            #     VALUES (?, ?, ?, '20120618 10:34:09 AM')""")
-            #     cursor.execute(sql_query, userId, 'Do It', 'Almost like Nike motto')   
-            #     logging.info('Executed the query')
-
-#UPDATE API method function
-def update(param1, param2):
+#PUT API method function
+def update(param1, param2, task_req_body):
     #connects to db
     cnxn = connect()
     cursor = cnxn.cursor()
-    logging.info('opened connection')        
-    logging.info('Going to execute update query')
     try:
-        # update task: title and description. Client passes userId and taskId, 
-        # for sprint 1 other fields are HARDCODED
-        sql_query = ("""UPDATE [dbo].[tasks]
-        SET title = 'Title updated by team1', description = 'Description updated by team1'
-        WHERE userId = ? AND taskId = ?""")   
-        rowcount = cursor.execute(sql_query, param1, param2).rowcount
-        logging.info(f"Executed the query: {rowcount} rows affected")
-#commits changes to db
+        logging.info('opened connection')        
+        logging.info('Going to execute UPDATE task query')

```

Comment:
What's your decision criteria for switching between info and debug?

Reply:
good catch, this is trailing from sprint1